ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/CustomExecutors.java

```diff
@@ -42,91 +41,156 @@
   }
 
-  public StatementExecutorResponse execute(
-      final ConfiguredStatement<?> statement,
-      final SessionProperties sessionProperties,
-      final KsqlExecutionContext executionCtx,
-      final ServiceContext serviceCtx
-  ) {
-    return executor.execute(
-        statement,
-        sessionProperties,
-        executionCtx,
-        serviceCtx
-        );
+  @VisibleForTesting
+  public StatementExecutor<TerminateQuery> terminateQuery() {
+    return (StatementExecutor<TerminateQuery>) EXECUTOR_MAP.get(TerminateQuery.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<DefineVariable> defineVariable() {
+    return (StatementExecutor<DefineVariable>) EXECUTOR_MAP.get(DefineVariable.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<UndefineVariable> undefineVariable() {
+    return (StatementExecutor<UndefineVariable>) EXECUTOR_MAP.get(UndefineVariable.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<SetProperty> setProperty() {
+    return (StatementExecutor<SetProperty>) EXECUTOR_MAP.get(SetProperty.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<UnsetProperty> unsetProperty() {
+    return (StatementExecutor<UnsetProperty>) EXECUTOR_MAP.get(UnsetProperty.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListProperties> listProperties() {
+    return (StatementExecutor<ListProperties>) EXECUTOR_MAP.get(ListProperties.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListStreams> listStreams() {
+    return (StatementExecutor<ListStreams>) EXECUTOR_MAP.get(ListStreams.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<DescribeStreams> describeStreams() {
+    return (StatementExecutor<DescribeStreams>) EXECUTOR_MAP.get(DescribeStreams.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListTables> listTables() {
+    return (StatementExecutor<ListTables>) EXECUTOR_MAP.get(ListTables.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<DescribeTables> describeTables() {
+    return (StatementExecutor<DescribeTables>) EXECUTOR_MAP.get(DescribeTables.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ShowColumns> showColumns() {
+    return (StatementExecutor<ShowColumns>) EXECUTOR_MAP.get(ShowColumns.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListFunctions> listFunctions() {
+    return (StatementExecutor<ListFunctions>) EXECUTOR_MAP.get(ListFunctions.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<DescribeFunction> describeFunction() {
+    return (StatementExecutor<DescribeFunction>) EXECUTOR_MAP.get(DescribeFunction.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListQueries> listQueries() {
+    return (StatementExecutor<ListQueries>) EXECUTOR_MAP.get(ListQueries.class);
+  }
+
+  @VisibleForTesting
+  public StatementExecutor<ListVariables> listVariables() {
```

Comment:
Do these test-only methods need to be public or is package-private sufficient?

Reply:
You are right! These methods can be package-private. Changed in https://github.com/confluentinc/ksql/pull/8480/commits/7ac1fe2c1b271aec0498c7f1315b34f0681bb7a4#diff-24e5d395473a57f7588579718e1f2209f549729a9d975c232ebedb3c0e52f905R116-R199.